tools/klockstat.py

```diff
@@ -373,6 +373,14 @@ def stack_id_err(stack_id):
 else:
     program += program_kprobe
 
+# Depending on whether DEBUG_LOCK_ALLOC is set, the proper kprobe may be either mutex_lock or mutex_lock_nested
+global kprobe_mutex_lock_func
+if not is_support_kfunc:
+    kprobe_mutex_lock_funcs = BPF.get_kprobe_functions(b'^mutex_lock(_nested)?$')
```

Comment:
Why regular expression here ?

Please refer to https://github.com/iovisor/bcc/commit/11614bcacdecd4d1f7015bb0f0311bb709207991

For kfunc, you should do the same check.

Reply:
Updated to follow the same convention as the linked commit.

Reply:
Please do the same for kfunc.

Reply:
Sorry, that I misunderstood you about the kfunc change. I didn't realize above that you were commenting on two separate changes that needed to be made.

So, I've done an updated version that I _think_ addresses the kfunc issue that you mentioned. However, because of the way the machines that I have access to define `bpf_trampoline_link_prog` none of my systems actually use the kfuncs, so I don't have a way to test it. Additionally, I'm relatively new to BPF stuff.

So the patch that I've updated on this MR is really just a guess, sorry if I get it wrong or took it the wrong way.